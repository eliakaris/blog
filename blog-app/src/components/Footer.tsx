import * as React from 'react';
import { NavLink } from 'react-router-dom';

function NavigationHeader() {
  return (
  <footer className="footer">
      <div className="container navbar">
        <hr />

        <div itemScope itemType="http://data-vocabulary.org/Person" className="hidden">
          <span itemProp="name">Elia Karagiannis</span>,
          <a href="http://www.eliakaris.com" itemProp="url">eliakaris.com</a>
          <span
              itemProp="address"
              itemScope={true}
              itemType="http://data-vocabulary.org/Address"
          >
            <span itemProp="locality">Seattle</span>, 
            <span itemProp="region">Washington</span>
          </span>
          <span itemProp="title">Software Development Engineer</span>
          <span itemProp="affiliation">Pinterest</span>
        </div>

        <p>© 2013 - 2017 <NavLink to="/about">Elia Karagiannis</NavLink>
         ∙ <a href="http://twitter.com/eliakaris">Follow @eliakaris</a>
          ∙ <a href="https://github.com/eliakaris/blog">View source</a>
        </p>
      </div>
    </footer>);
}

export default NavigationHeader;