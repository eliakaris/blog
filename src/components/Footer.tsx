import * as React from 'react';

function NavigationHeader() {
  return (
    <footer>
      <div itemScope itemType="http://data-vocabulary.org/Person" className="hidden">
        <span itemProp="name">Elia Karagiannis</span>,
        <a href="http://www.eliakaris.com" itemProp="url">eliakaris.com</a>
        <span
            itemProp="address"
            itemScope={true}
            itemType="http://data-vocabulary.org/Address">
          <span itemProp="locality">Seattle</span>, 
          <span itemProp="region">Washington</span>
        </span>
        <span itemProp="title">Software Development Engineer</span>
        <span itemProp="affiliation">Pinterest</span>
      </div>
      <div className="container">
        <div className="row">
          <div className="col-lg-8 col-md-10 mx-auto">
            <ul className="list-inline text-center">
              <li className="list-inline-item">
                <a href="http://twitter.com/eliakaris">
                  <span className="fa-stack fa-lg">
                    <i className="fa fa-circle fa-stack-2x"></i>
                    <i className="fa fa-twitter fa-stack-1x fa-inverse"></i>
                  </span>
                </a>
              </li>
              <li className="list-inline-item">
                <a href="https://github.com/eliakaris/blog">
                  <span className="fa-stack fa-lg">
                    <i className="fa fa-circle fa-stack-2x"></i>
                    <i className="fa fa-github fa-stack-1x fa-inverse"></i>
                  </span>
                </a>
              </li>
            </ul>
            <p className="copyright text-muted">Copyright Â© Elia Karagiannis 2017</p>
          </div>
        </div>
      </div>
    </footer>);
}

export default NavigationHeader;