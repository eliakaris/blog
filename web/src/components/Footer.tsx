import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTwitter, faGithub } from '@fortawesome/free-brands-svg-icons';

function Footer() {
  return (
    <footer>
      <div itemScope itemType="http://data-vocabulary.org/Person" className="hidden">
        <span itemProp="name">Elia Karagiannis</span>,
        <a href="http://www.eliakaris.com" itemProp="url">eliakaris.com</a>
        <span
            itemProp="address"
            itemScope={true}
            itemType="http://data-vocabulary.org/Address"
        >
          <span itemProp="locality">Seattle</span>, 
          <span itemProp="region">Washington</span>
        </span>
        <span itemProp="title">Software Development Engineer</span>
        <span itemProp="affiliation">Pinterest</span>
      </div>
      <div className="container">
        <div className="row">
          <div className="col-lg-8 col-md-10 mx-auto">
            <ul className="list-inline text-center">
              <li className="list-inline-item">
                <a href="http://twitter.com/eliakaris">
                  <FontAwesomeIcon icon={faTwitter} size="2x" />
                </a>
              </li>
              <li className="list-inline-item">
                <a href="https://github.com/eliakaris/blog">
                  <FontAwesomeIcon icon={faGithub} size="2x" />
                </a>
              </li>
            </ul>
            <p className="copyright text-muted">Copyright Â© Elia Karagiannis 2019</p>
          </div>
        </div>
      </div>
    </footer>);
}

export default Footer;
