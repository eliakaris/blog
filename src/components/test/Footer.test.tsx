import * as React from 'react';
import Footer from '../Footer';
import * as ShallowRenderer from 'react-test-renderer/shallow';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';

configure({ adapter: new ReactSixteenAdapter() });

describe('<Footer /> tests', () => {
  let renderer: any;

  beforeAll(() => {
    renderer = ShallowRenderer.createRenderer();
  });

  it('renders a snapshot', () => {
    const component = renderer.render(<Footer />);
    expect(component).toMatchSnapshot();
  });
});