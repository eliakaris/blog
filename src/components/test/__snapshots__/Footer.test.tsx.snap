// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> tests renders a snapshot 1`] = `
<footer>
  <div
    className="hidden"
    itemScope={true}
    itemType="http://data-vocabulary.org/Person"
  >
    <span
      itemProp="name"
    >
      Elia Karagiannis
    </span>
    ,
    <a
      href="http://www.eliakaris.com"
      itemProp="url"
    >
      eliakaris.com
    </a>
    <span
      itemProp="address"
      itemScope={true}
      itemType="http://data-vocabulary.org/Address"
    >
      <span
        itemProp="locality"
      >
        Seattle
      </span>
      ,
      <span
        itemProp="region"
      >
        Washington
      </span>
    </span>
    <span
      itemProp="title"
    >
      Software Development Engineer
    </span>
    <span
      itemProp="affiliation"
    >
      Pinterest
    </span>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-8 col-md-10 mx-auto"
      >
        <ul
          className="list-inline text-center"
        >
          <li
            className="list-inline-item"
          >
            <a
              href="http://twitter.com/eliakaris"
            >
              <span
                className="fa-stack fa-lg"
              >
                <i
                  className="fa fa-circle fa-stack-2x"
                />
                <i
                  className="fa fa-twitter fa-stack-1x fa-inverse"
                />
              </span>
            </a>
          </li>
          <li
            className="list-inline-item"
          >
            <a
              href="https://github.com/eliakaris/blog"
            >
              <span
                className="fa-stack fa-lg"
              >
                <i
                  className="fa fa-circle fa-stack-2x"
                />
                <i
                  className="fa fa-github fa-stack-1x fa-inverse"
                />
              </span>
            </a>
          </li>
        </ul>
        <p
          className="copyright text-muted"
        >
          Copyright Â© Elia Karagiannis 2018
        </p>
      </div>
    </div>
  </div>
</footer>
`;
